src/common/utils/use-waiting-message.ts

```diff
@@ -0,0 +1,48 @@
+import { useState } from 'react';
+
+// Keys are the seconds to wait before showing the message
+export type WaitingMessages = {
+  [key: string]: string;
+};
```

Comment:
```suggestion
export type WaitingMessages = Record<string, string>;
```

Reply:
Though, couldn't this be number, string if it refers to time?

Reply:
I hesitated using number since anyway the keys are stored in string in Javascript objects, but here indeed adding number in Typescript can help with documentation/readability. I just changed it. 👍

Reply:
Wasn't sure if it'd work, but great